spring-rabbit/src/main/java/org/springframework/amqp/rabbit/log4j/AmqpAppender.java

```diff
@@ -409,6 +409,8 @@ public void setCharset(String charset) {
 	 */
 	protected void startSenders() {
 		senderPool = Executors.newCachedThreadPool();
+		synchronized(this) {
+		} // (logically) flush all variables to main memory
```

Comment:
My initial fix was to make all the fields volatile but then copy them into the `EventSender` via a ctor, as `final` fields, to avoid accessing the volatile fields at runtime.

I thought "there has to be a better way than this". So I came up with this after a bit more thought. I checked the bytecode to make sure the compiler didn't optimize out the block with no code but I figured it wouldn't because `synchronized` has these semantics.

This technique ok here because the fields are only set up once, during log initialization.